webpack/assets/javascripts/jquery.ui.custom_spinners.js

```diff
@@ -0,0 +1,111 @@
+$(function () {
+  $.widget("ui.limited_spinner", $.ui.spinner, {
+    options: {
+      soft_maximum: 0,
+      error_target: null
+    },
+    _validate: function() {
+      if (this.options['soft_maximum'] != 0) {
+        if (this.value() > this.options['soft_maximum']) {
+          this.options["error_target"].show();
+        } else {
+          this.options["error_target"].hide();
+        }
+      }
+    },
+    _spin: function (step, event) {
+      var result = this._super(step, event);
+      this._validate();
+      return result;
+    }
+  });
+
+  $.widget("ui.byte_spinner", $.ui.limited_spinner, {
+    options: {
+      step: 1,
+      min: 1,
+      incremental: false,
+      value_target: null
+    },
+    _megabyte: 1024 * 1024,
```

Comment:
these could be set in the module context as `const megabyte = 1024 **2`. Since this is a module that won't pollute the global namespace and save all of the annoying `this._megabyte` etc